java/arcs/core/common/Id.kt

```diff
@@ -96,12 +96,13 @@ interface Id {
 }
 
 /** Convenience to parse a [String] into an [ArcId]. */
-fun String.toArcId(): ArcId = toId().let { ArcId(it.root, it.idTree) }
+fun String.toArcId(): ArcId = toId().let { ArcId(it.root, it.idTree, this) }
 
 /** [Id] for an Arc. */
 data class ArcId /* internal */ constructor(
     override val root: String,
-    override val idTree: List<String>
+    override val idTree: List<String>,
+    val name: String
```

Comment:
What is the name? (provide KDoc here?)

Reply:
is name intended to be RecipeName/PlanName?